### ISSUE TITLE
Overly strict parsing of content-type in metadata indexer

### ISSUE THREAD

**Issue by kislyuk** [HumanCellAtlas/data-store#630]
@sampierson encountered this problem when ingesting a bundle through the blue box. The indexer requires the content-type to be exactly "application/json", but modifiers like "application/json; foo=bar" should be admissible.

@Bento007 interest

```
>aegea logs /aws/lambda/dss-index-staging --start-time=-1h| grep 3153fc1e-4136-454d-a67d-24a0c73cd632
2017-11-08 16:04:16+00:00 Domovoi dispatch of event {'Records': [{'EventSource': 'aws:sns', 'EventVersion': '1.0', 'EventSubscriptionArn': 'arn:aws:sns:us-east-1:861229788715:domovoi-s3-bucket-events-org-humancellatlas-dss-staging:5b7a3f50-02aa-4d00-a3a5-45a1ae1ea9a2', 'Sns': {'Type': 'Notification', 'MessageId': '49b2aecb-aacc-51b2-9402-fcb6878c6e45', 'TopicArn': 'arn:aws:sns:us-east-1:861229788715:domovoi-s3-bucket-events-org-humancellatlas-dss-staging', 'Subject': 'Amazon S3 Notification', 'Message': '{"Records":[{"eventVersion":"2.0","eventSource":"aws:s3","awsRegion":"us-east-1","eventTime":"2017-11-08T16:04:16.536Z","eventName":"ObjectCreated:Put","userIdentity":{"principalId":"AWS:AROAIECL3JT6IRTPBWHTM:dss-staging"},"requestParameters":{"sourceIPAddress":"34.230.83.28"},"responseElements":{"x-amz-request-id":"651B54618AB03AD5","x-amz-id-2":"CrH9P8YQulidT7krnHYKUXrOEEuhv54xPxjDTqMBOqolkN01tCqARFWBM9lVS+n3phylAZWKJW0="},"s3":{"s3SchemaVersion":"1.0","configurationId":"YzQ4NTE0MTItYjMwZS00Mjg2LTg4MTEtOTUyYjQyNmExNWE0","bucket":{"name":"org-humancellatlas-dss-staging","ownerIdentity":{"principalId":"A29PZ5XRQWJUUM"},"arn":"arn:aws:s3:::org-humancellatlas-dss-staging"},"object":{"key":"bundles/3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z","size":2111,"eTag":"62c7466979b6401da49fc34ab20ecbbf","sequencer":"005A032B0078D5F101"}}}]}', 'Timestamp': '2017-11-08T16:04:16.620Z', 'SignatureVersion': '1', 'Signature': 'amCvGbgjfR24t62/fQB9jOoB4JdLF31CDT6Y7UwyJ2xaJbGGg2WeDjZEnE/Dd24T4wLOk3C95B3veYms9G1tNtcroW2GMFQTfz3PvqcdfeOKaREiItTsgSHo4QOd3SbC+ZzVMEQX1F4et/NSZxmsrFa18/aaT82y3DI0hO/2ndSLnvpJOWzS9x4L9E9y1pEAFYHsfwDd3AKkDhEKEEV5V/yjBuhFfkFntfKpJDnbcI9yN0EDczanH+8eRjJF0+LrqBtpxyA/AlMSioeLkzcEkoToivubU+PtmpQ+BIvvGVInnqqIdSLDhVJFlmIBFaVoQnh0EVkqjHzbTA0iYR41jA==', 'SigningCertUrl': 'https://sns.us-east-1.amazonaws.com/SimpleNotificationService-433026a4050d206028891664da859041.pem', 'UnsubscribeUrl': 'https://sns.us-east-1.amazonaws.com/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:us-east-1:861229788715:domovoi-s3-bucket-events-org-humancellatlas-dss-staging:5b7a3f50-02aa-4d00-a3a5-45a1ae1ea9a2', 'MessageAttributes': {}}}]}
2017-11-08 16:04:16+00:00 Domovoi - INFO - Received aws creation event for bundle which will be indexed: bundles/3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z
2017-11-08 16:04:16+00:00 Domovoi - DEBUG - Read bundle manifest from bucket org-humancellatlas-dss-staging with bundle key bundles/3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z: {"format": "0.0.1", "version": "2017-11-08T160415.775304Z", "files": [{"name": "project.json", "uuid": "a80598b7-d7c5-4bbd-a030-6ff472e17bdb", "version": "2017-11-08T160407.272102Z", "content-type": "application/json; dcp-type=\"metadata/project\"", "size": 3312, "indexed": true, "crc32c": "f9f8f7c1", "s3-etag": "36c1eb2f1937bdfdb1c8bda8a1bf0f8f", "sha1": "300b625a47e643ab72b37da4dd68f09bfec5794f", "sha256": "7393765a1cfd863a9d46f7a6cc2c7ff867bab92a0b4f5ac3c1c5c6ef7591308a"}, {"name": "sample.json", "uuid": "b5abf0e6-62bc-4967-a550-79cbfd519510", "version": "2017-11-08T160409.096206Z", "content-type": "application/json; dcp-type=\"metadata/sample\"", "size": 1247, "indexed": true, "crc32c": "51c577d5", "s3-etag": "f3ddd9234445f5b296b8025b5b1854c5", "sha1": "622c24e90305527b4a99ab7481ec408d30ed2224", "sha256": "32726da906dc0094eb69bff9508721ba75960872cff204c941dd7febcb781561"}, {"name": "R1.fastq.gz", "uuid": "4fe297f4-60e9-49c4-8855-6bcd2eaa6610", "version": "2017-11-08T160410.236583Z", "content-type": "application/gzip; dcp-type=data", "size": 125191, "indexed": false, "crc32c": "4ef74578", "s3-etag": "c7bbee4c46bbf29432862e05830c8f39", "sha1": "17f8b4be0cc6e8281a402bb365b1283b458906a3", "sha256": "fe6d4fdfea2ff1df97500dcfe7085ac3abfb760026bff75a34c20fb97a4b2b29"}, {"name": "R2.fastq.gz", "uuid": "9be5d0d9-225b-49b4-a31f-1f5514e46dcd", "version": "2017-11-08T160411.161628Z", "content-type": "application/gzip; dcp-type=data", "size": 130024, "indexed": false, "crc32c": "69987b3e", "s3-etag": "a3a9f23d07cfc5e40a4c3a8adf3903ae", "sha1": "f166b6952e30a41e1409e7fb0cb0fb1ad93f3f21", "sha256": "c305bee37b3c3735585e11306272b6ab085f04cd22ea8703957b4503488cfeba"}, {"name": "assay.json", "uuid": "8ad5e57b-dff9-4774-828f-fb76b8da1a1a", "version": "2017-11-08T160414.028878Z", "content-type": "application/json; dcp-type=\"metadata/assay\"", "size": 821, "indexed": true, "crc32c": "9b7192a9", "s3-etag": "c74a1c7afc80c639c59d0f265bb32e0a", "sha1": "3ff295f27c1c52ceb2f82eac7319582c34697f79", "sha256": "e9faa94de46940a2bd33fb9420d9672bab28c01a14cc394775ef133e92c24f33"}], "creator_uid": 8008}
2017-11-08 16:04:16+00:00 Domovoi - WARNING - In bundle 3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z the file "project.json" is marked for indexing yet has content type "application/json; dcp-type="metadata/project"" instead of the required content type "application/json". This file will not be indexed.
2017-11-08 16:04:16+00:00 Domovoi - WARNING - In bundle 3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z the file "sample.json" is marked for indexing yet has content type "application/json; dcp-type="metadata/sample"" instead of the required content type "application/json". This file will not be indexed.
2017-11-08 16:04:16+00:00 Domovoi - WARNING - In bundle 3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z the file "assay.json" is marked for indexing yet has content type "application/json; dcp-type="metadata/assay"" instead of the required content type "application/json". This file will not be indexed.
2017-11-08 16:04:17+00:00 [INFO]	2017-11-08T16:04:17.87Z	78acb29f-c49e-11e7-b5de-19c2f60c698c	PUT https://search-dss-index-dev-wc7kz2vuj7mpfjbxnp4ftplpuu.us-east-1.es.amazonaws.com:443/dss-docs-aws-staging/doc/3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z [status:201 request:0.152s]
2017-11-08 16:04:20+00:00 Domovoi dispatch of event {'Records': [{'EventSource': 'aws:sns', 'EventVersion': '1.0', 'EventSubscriptionArn': 'arn:aws:sns:us-east-1:861229788715:dss-gs-bucket-events-org-humancellatlas-dss-staging:f3bd427f-5e9d-44a8-8108-1fdceb5320c5', 'Sns': {'Type': 'Notification', 'MessageId': '51f6c38f-656a-523c-8508-91f2482e0ccb', 'TopicArn': 'arn:aws:sns:us-east-1:861229788715:dss-gs-bucket-events-org-humancellatlas-dss-staging', 'Subject': None, 'Message': '{"timestamp":"2017-11-08T16:04:17.501Z","eventType":"providers/cloud.storage/eventTypes/object.change","resource":"projects/_/buckets/org-humancellatlas-dss-staging/objects/bundles/3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z#1510157057501062","data":{"kind":"storage#object","resourceState":"exists","id":"org-humancellatlas-dss-staging/bundles/3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z/1510157057501062","selfLink":"https://www.googleapis.com/storage/v1/b/org-humancellatlas-dss-staging/o/bundles%2F3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z","name":"bundles/3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z","bucket":"org-humancellatlas-dss-staging","generation":"1510157057501062","metageneration":"1","contentType":"application/octet-stream","timeCreated":"2017-11-08T16:04:17.355Z","updated":"2017-11-08T16:04:17.355Z","storageClass":"STANDARD","size":"2111","md5Hash":"NjJjNzQ2Njk3OWI2NDAxZGE0OWZjMzRhYjIwZWNiYmY=","mediaLink":"https://www.googleapis.com/download/storage/v1/b/org-humancellatlas-dss-staging/o/bundles%2F3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z?generation=1510157057501062&alt=media","crc32c":"zMZc2w=="}}', 'Timestamp': '2017-11-08T16:04:20.131Z', 'SignatureVersion': '1', 'Signature': 'NWw2MkxhUQtZENerx7KjG1PoVCY05Dju1SLENWKPefr3uZxSlzB3Uwq0IGWvfInE1kyfHxxFnWgnDClreLfxG9SEeU4ra5Y2pScZEJqoIKbhYgDDDMceUAlzopKPOOVNF0kZI1usFbQwV8qR8fYjk4XzHZPsKEpAAVNc0ZX0U2tWtgRoBrs7czUDahdg06K/P1C57WWrRbkAT9eMzrJKBRKCmr3ZdJmfTwcOjXpueiGAhN/cpZHlNtGnIWknjchfaV0e9LO5/E3CI8KYUnwe61nFbeo4PY9SXi+qRdwh2cV51F/z1N7i2jkezl0g2/IOlrl1qPO5APgdTJXU8IY9ug==', 'SigningCertUrl': 'https://sns.us-east-1.amazonaws.com/SimpleNotificationService-433026a4050d206028891664da859041.pem', 'UnsubscribeUrl': 'https://sns.us-east-1.amazonaws.com/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:us-east-1:861229788715:dss-gs-bucket-events-org-humancellatlas-dss-staging:f3bd427f-5e9d-44a8-8108-1fdceb5320c5', 'MessageAttributes': {}}}]}
2017-11-08 16:04:20+00:00 Domovoi - INFO - Received gcp creation event for bundle which will be indexed: bundles/3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z
2017-11-08 16:04:20+00:00 Domovoi - WARNING - Failed notification for subscription 84d9a3d3-d2df-465f-8f73-a3ff533f9723 for bundle 3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z with transaction id 97ed104b-51f2-447e-844b-1b498073b3c0 Code: 404
2017-11-08 16:04:20+00:00 Domovoi - DEBUG - Finished index processing of aws creation event for bundle: bundles/3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z
2017-11-08 16:04:20+00:00 Domovoi - DEBUG - Read bundle manifest from bucket org-humancellatlas-dss-staging with bundle key bundles/3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z: {"format": "0.0.1", "version": "2017-11-08T160415.775304Z", "files": [{"name": "project.json", "uuid": "a80598b7-d7c5-4bbd-a030-6ff472e17bdb", "version": "2017-11-08T160407.272102Z", "content-type": "application/json; dcp-type=\"metadata/project\"", "size": 3312, "indexed": true, "crc32c": "f9f8f7c1", "s3-etag": "36c1eb2f1937bdfdb1c8bda8a1bf0f8f", "sha1": "300b625a47e643ab72b37da4dd68f09bfec5794f", "sha256": "7393765a1cfd863a9d46f7a6cc2c7ff867bab92a0b4f5ac3c1c5c6ef7591308a"}, {"name": "sample.json", "uuid": "b5abf0e6-62bc-4967-a550-79cbfd519510", "version": "2017-11-08T160409.096206Z", "content-type": "application/json; dcp-type=\"metadata/sample\"", "size": 1247, "indexed": true, "crc32c": "51c577d5", "s3-etag": "f3ddd9234445f5b296b8025b5b1854c5", "sha1": "622c24e90305527b4a99ab7481ec408d30ed2224", "sha256": "32726da906dc0094eb69bff9508721ba75960872cff204c941dd7febcb781561"}, {"name": "R1.fastq.gz", "uuid": "4fe297f4-60e9-49c4-8855-6bcd2eaa6610", "version": "2017-11-08T160410.236583Z", "content-type": "application/gzip; dcp-type=data", "size": 125191, "indexed": false, "crc32c": "4ef74578", "s3-etag": "c7bbee4c46bbf29432862e05830c8f39", "sha1": "17f8b4be0cc6e8281a402bb365b1283b458906a3", "sha256": "fe6d4fdfea2ff1df97500dcfe7085ac3abfb760026bff75a34c20fb97a4b2b29"}, {"name": "R2.fastq.gz", "uuid": "9be5d0d9-225b-49b4-a31f-1f5514e46dcd", "version": "2017-11-08T160411.161628Z", "content-type": "application/gzip; dcp-type=data", "size": 130024, "indexed": false, "crc32c": "69987b3e", "s3-etag": "a3a9f23d07cfc5e40a4c3a8adf3903ae", "sha1": "f166b6952e30a41e1409e7fb0cb0fb1ad93f3f21", "sha256": "c305bee37b3c3735585e11306272b6ab085f04cd22ea8703957b4503488cfeba"}, {"name": "assay.json", "uuid": "8ad5e57b-dff9-4774-828f-fb76b8da1a1a", "version": "2017-11-08T160414.028878Z", "content-type": "application/json; dcp-type=\"metadata/assay\"", "size": 821, "indexed": true, "crc32c": "9b7192a9", "s3-etag": "c74a1c7afc80c639c59d0f265bb32e0a", "sha1": "3ff295f27c1c52ceb2f82eac7319582c34697f79", "sha256": "e9faa94de46940a2bd33fb9420d9672bab28c01a14cc394775ef133e92c24f33"}], "creator_uid": 8008}
2017-11-08 16:04:20+00:00 Domovoi - WARNING - In bundle 3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z the file "project.json" is marked for indexing yet has content type "application/json; dcp-type="metadata/project"" instead of the required content type "application/json". This file will not be indexed.
2017-11-08 16:04:20+00:00 Domovoi - WARNING - In bundle 3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z the file "sample.json" is marked for indexing yet has content type "application/json; dcp-type="metadata/sample"" instead of the required content type "application/json". This file will not be indexed.
2017-11-08 16:04:20+00:00 Domovoi - WARNING - In bundle 3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z the file "assay.json" is marked for indexing yet has content type "application/json; dcp-type="metadata/assay"" instead of the required content type "application/json". This file will not be indexed.
2017-11-08 16:04:21+00:00 [INFO]	2017-11-08T16:04:21.114Z	7ac447fb-c49e-11e7-a47a-8d1cf176c5ea	PUT https://search-dss-index-dev-wc7kz2vuj7mpfjbxnp4ftplpuu.us-east-1.es.amazonaws.com:443/dss-docs-gcp-staging/doc/3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z [status:201 request:0.229s]
2017-11-08 16:04:21+00:00 Domovoi - DEBUG - Finished index processing of gcp creation event for bundle: bundles/3153fc1e-4136-454d-a67d-24a0c73cd632.2017-11-08T160415.775304Z
```

**Comment by ttung** [comment]
I think this should just rely on the indexed field.  No reason we should even look at the content-type.  See #629

**Comment by Bento007** [comment]
was there a reason for this initially?

**Comment by kislyuk** [comment]
Fixed in #631.
